modules/drivers/presto-jdbc/src/metabase/driver/presto_jdbc.clj

```diff
@@ -612,6 +612,14 @@
   [_driver _database _table]
   nil)
 
+(defmethod driver/can-connect? :presto-jdbc
+  [driver {:keys [catalog], :as details}]
+  (and ((get-method driver/can-connect? :sql-jdbc) driver details)
+       (sql-jdbc.conn/with-connection-spec-for-testing-connection [spec [driver details]]
+         ;; jdbc/query is used to see if we throw, we want to ignore the results
+         (jdbc/query spec (format "SHOW SCHEMAS FROM %s" catalog))
```

Comment:
This fixes a bug with `driver/can-connect?` for presto-jdbc. It will now throw if the catalog doesn't exist.